lib/core/src/Cardano/Wallet/Primitive/Types/TokenQuantity.hs

```diff
@@ -178,8 +178,9 @@ equipartition q =
     fmap TokenQuantity . equipartitionNatural (unTokenQuantity q)
 
 -- | Partitions a token quantity into a number of parts, where the size of each
---   part is proportional to the size of its corresponding element in the given
---   list of weights, and the number of parts is equal to the number of weights.
+--   part is proportional (modulo rounding) to the size of its corresponding
```

Comment:
With respect to rounding, `voting-tools` will assign any remainder to the last element, is that something you could consider here?

Reply:
It makes writing laws easier and prevents a poor choice of weights compromising that `valueIn == valueOut` law.

Reply:
> With respect to rounding, voting-tools will assign any remainder to the last element, is that something you could consider here?

In this case, we want to avoid biasing any particular element in the result.

So in the (rare) case that all weights are zero, we split the target quantity as evenly as we can across the output (modulo rounding).

Note that this method _still_ has some unavoidable bias, because in the event that the target value is not evenly divisible by the number of parts, we distribute any excess to the tail of the list. (This follows from the behaviour of `partitionNatural` in `Numeric.Util`.